2015/proposals/agarwal-rajat.md

```diff
@@ -0,0 +1,92 @@
+
+2. Deciding on the best name
+From the possible names returned from different services, an algorithm to determine the best name has to be developed. This would include giving different weights to parameters like reliability of source, consistency with other services and other parameters which need to be thought of. 
+
+3. Updating the database
+Once the decision about the best name is made, this component will update the database with the best name and its location.
+
+4. Building control flow
+This component would allow the user to choose whether to run name resolution or not. Also, name resolution for the species should not run if the species name is not present in the database. This will act as a bridge between the UI-Database module and the Name Resolution module.
+
+
+## Schedule of Deliverables
+
+Before the coding period begins, I intend to do the following:
+1. Get more familiar with the existing code base.
+2. Finding the best name resolution services which are to be used.
+3. Making the design decisions of various modules with the mentors.
+
+### May 25th -  June 7th
+
+Getting the list of names, by interfacing with the different APIs, that are to be given as input to the best name deciding algorithm.  
+
+### June 8th - June 21st
+
+Developing the best name deciding algorithm. 
+
+### June 22nd - July 5th
+
+Updating the database with the best name.
+
+### July 6th - July 19th
+
+Building the control flow option. 
+
+### July 20th - August 2rd
+
+Complete the control flow and start fixing bugs.
+
+### August 3rd - August 16th
+
+Make the code more efficient and robust with documentation. Fix more bugs.
+
+### August 17th - August 21st 19:00 UTC
+
+Week to scrub code, write tests, improve documentation, etc.
+
+## Future works
+
+1. Adding more reliable name resolution APIs.
+2. Improving the accuracy the best name deciding algorithm.
+3. Bug fixing the existing and adding more engines for different formats of data. For example - JSON, XML etc.
+4. Adding more features to the user interface like giving the option to the user to update the names from a specific source.
+5. Support for the existing code base.
+
+## Open Source Development Experience
+
+I'm afraid that I have no experience contributing to open source projects (this would be my first!) but have used many open source tools and modified their code to accomplish different tasks. This has also involved reading and getting familiar with moderate sized code bases and development practices.
+
+I am currently working with Prof. Sharad Goel from Stanford University doing research in Data Science and collaborating with other researchers using GitHub.
+
+I have also worked with a start-up from MIT Media Lab which involved modifying and tweaking open source tools and collaborating with other developers in the team using GitHub (The repository is private).
+
+I also like to use git for version control and put up any projects that I do on my GitHub account.
```

Comment:
"I also like to use git for version control" -> "I also like to use Git for version control"